mysql-test/t/percona_heap_bug_ps296.test

```diff
@@ -0,0 +1,36 @@
+
+
+--disable_query_log
+--disable_result_log
+create table a (id int unsigned not null auto_increment primary key);
+delimiter |;
+insert into a () values ();
+CREATE PROCEDURE dowhile()
+BEGIN
+ DECLARE v1 INT DEFAULT 1;
+ while v1 < 15 do 
+insert into a select null from a;
+ set v1 = v1 + 1;
+ end while;
+ END;
+|
+delimiter ;|
+call dowhile();
+CREATE TABLE `big1` (`id` varbinary(999) NOT NULL,PRIMARY KEY (`id`)) ENGINE=InnoDB;
+CREATE TABLE `big2` (`id` varbinary(999) NOT NULL, PRIMARY KEY (`id`)) ENGINE=InnoDB;
+insert into big1 select * from a;
+insert into big2 select * from a;
+set session optimizer_switch='duplicateweedout=off,materialization=on';
+--enable_result_log
+--enable_query_log
+
+select * from big1 where id in (select id from big2 ignore index (primary)) order by id asc limit 10;
```

Comment:
so this is due to a temp MEMORY table that stores VARBINARY. How do we end up getting one? Dynamic varchar patch did not enable them for the query optimizer use. Does the upstream use MEMORY table too?

Reply:
Correction: the explain shows "filesort" in the test case, so the memory table is converted into an on disk table later. But at the point where the decision is made, it selects a memory table - and because of this, it uses the memory table's key length in the calculation too.